src/pages/index.js

```diff
@@ -135,20 +304,83 @@ document.querySelectorAll(".modal").forEach((modal) => {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+
+  const submitButton = editFormElement.querySelector(".modal__submit-btn");
+
+  // Set loading state
+  submitButton.disabled = true;
+  submitButton.classList.add("modal__submit-btn_loading");
+  submitButton.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+
+      // Reset form and close modal
+      editFormElement.reset();
+      closeModal(editModal);
+
+      // Reset validation state
+      resetValidation(editFormElement, settings);
+
+      // Reset button state
+      submitButton.classList.remove("modal__submit-btn_loading");
+      submitButton.textContent = "Save";
+      disableButton(submitButton, settings);
+    })
+    .catch((err) => {
+      console.error("Profile update failed:", err);
+
+      // Reset button state but keep current validation
+      submitButton.classList.remove("modal__submit-btn_loading");
+      submitButton.textContent = "Save";
+      submitButton.disabled = false;
+    });
 }
 
-function handleAddCardSubmit(evt) {
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
   evt.preventDefault();
-  const inputValue = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValue);
-  cardsList.prepend(cardElement);
 
-  cardForm.reset();
-  disableButton(cardSubmitBtn, settings);
-  closeModal(cardModal);
+  api
```

Comment:
When you delete a card, the button text should display `Deleting...`, according to the design